internal/engine/client.go

```diff
@@ -95,5 +95,6 @@ func waitBuildkit(ctx context.Context, host string) error {
 		}
 		time.Sleep(retryPeriod)
 	}
-	return errors.New("buildkit failed to respond")
+	return errors.New("buildkit failed to respond." +
+		"If engine was manually stopped, run it again or delete it")
```

Comment:
This message is very specific to the default path where we start the engine on the users behalf in docker, but it would be misleading for the other connection types like unix, tcp.

I think a better approach to this would be to, in the retry loop above this, save non-nil errors received from the `ListWorkers` call. If the retries run out, print the most recent error to stderr.

The Go and Python SDKs both will currently collect stderr output and send return it as an error to the caller if the connect fails. We would thus just need to update the Node SDK to do that too.

This will result in error messages being returned to the client that are more descriptive and specific to the connection type. I just tried it w/ a stopped docker container and got the error message `Container aa1fcdbe229ed2b027f10bde7a9664145d5ee60bc1400d55e54f8597bf803a40 is not running`. I also tried it with `docker-container://` where the container didn't exist and got `No such container: test-dagger-engine`.